tests/test_admin_utils.py

```diff
@@ -0,0 +1,244 @@
+        )
+
+        platform_obj = (
+            store.session.query(store.db_classes.Platform)
+            .filter(store.db_classes.Platform.name == "Platform1")
+            .all()[0]
+        )
+
+        query = make_query_for_cols(
+            store.db_classes.Platform, platform_obj, ["identifier", "name"], store.session
+        )
+
+        query_str = str(query.statement.compile(compile_kwargs={"literal_binds": True}))
+
+        # Name field was specified, so should be in a WHERE clause
+        assert "\"Platforms\".name = 'Platform1'" in query_str
+
+        # Identifier field was specified, so should be there too
+        assert "\"Platforms\".identifier = '123'" in query_str
+
+        # Shouldn't be searching for nationality as it's not in the list
+        assert (
+            f"\"Platforms\".nationality_id = '{remove_dashes(platform_obj.nationality_id)}'"
+            not in query_str
+        )
+
+
+def test_make_query_for_all_data_columns():
+    store = DataStore("", "", "", 0, db_name=":memory:", db_type="sqlite")
+    store.initialise()
+
+    with store.session_scope():
+        change_id = store.add_to_changes("TEST", datetime.utcnow(), "TEST").change_id
+        store.add_to_platform_types("PlatformType1", change_id)
+        store.add_to_nationalities("UK", change_id)
+        store.add_to_privacies("Private", 0, change_id)
+        store.add_to_platforms(
+            "Platform1", "123", "UK", "PlatformType1", "Private", change_id=change_id
+        )
+        store.add_to_platforms(
+            "Platform2", "123", "UK", "PlatformType1", "Private", change_id=change_id
+        )
+
+        platform_obj = (
+            store.session.query(store.db_classes.Platform)
+            .filter(store.db_classes.Platform.name == "Platform1")
+            .all()[0]
+        )
+
+        query = make_query_for_all_data_columns(
+            store.db_classes.Platform, platform_obj, store.session
+        )
+
+        query_str = str(query.statement.compile(compile_kwargs={"literal_binds": True}))
+
+        # Check the right WHERE clauses are added
+        assert "\"Platforms\".name = 'Platform1'" in query_str
+        assert (
+            f"\"Platforms\".nationality_id = '{remove_dashes(platform_obj.nationality_id)}'"
+            in query_str
+        )
+        assert (
+            f"\"Platforms\".platform_type_id = '{remove_dashes(platform_obj.platform_type_id)}'"
+            in query_str
+        )
+
+        # Check NULL fields are ignored
+        assert '"Platforms".identifier IS NULL' not in query_str
+
+        # Check created_date isn't present in a WHERE clause
+        assert "created_date =" not in query_str
+
+        # Check we're not searching for a privacy id
+        assert (
+            f"\"Platforms\".privacy_id = '{remove_dashes(platform_obj.privacy_id)}'"
+            not in query_str
+        )
+
+
+def test_table_name_to_class_name():
```

Comment:
I also implemented a test which checks all table names:
https://github.com/debrief/pepys-import/blob/dd6fe1bf50e37a7d3477335b34500ba5e160c952/tests/test_table_name_utils.py#L51-L57

Reply:
Great, if we all use the same function then we can just use your set of tests.